packages/copyable/src/Copyable.tsx

```diff
@@ -0,0 +1,283 @@
+  font-size: 14px;
+  line-height: 20px;
+  margin-bottom: 6px;
+`;
+
+const containerStyle = css`
+  position: relative;
+  height: 48px;
+  width: 400px;
+  margin: 1px 0;
+  overflow: hidden;
+`;
+
+const copyableContainerStyle = css`
+  height: 36px;
+`;
+
+const codeStyle = css`
+  display: inline-flex;
+  align-items: center;
+
+  height: 100%;
+  width: 100%;
+  padding-left: 12px;
+  border: 1px solid;
+  border-radius: 4px;
+
+  font-size: 14px;
+  overflow: hidden;
+`;
+
+const largeCodeStyle = css`
+  font-size: 16px;
+`;
+
+const buttonWrapperStyle = css`
+  position: absolute;
+  z-index: 1;
+  right: 0;
+  top: 0;
+  height: 100%;
+`;
+
+const buttonStyle = css`
+  height: 100%;
+  border-top-left-radius: 0;
+  border-bottom-left-radius: 0;
+  filter: drop-shadow(2px 0 6px ${transparentize(0.7, uiColors.gray.dark1)});
+`;
+
+const iconStyle = css`
+  padding-right: 6px;
+`;
+
+interface CopyableProps {
+  variant?: Variant;
+  children: string;
+  label: string;
+  description?: string;
+  copyable?: boolean;
+  size?: Size;
+}
+
+export default function Copyable({
+  variant = Variant.Default,
+  children,
+  label,
+  description,
+  copyable = true,
+  size = Size.Default,
+}: CopyableProps) {
+  const colorSet = colorSets[variant];
+
+  const [copied, setCopied] = useState(false);
+  const [showCopyButton, setShowCopyButton] = useState(false);
+  const [buttonRef, setButtonRef] = useState<HTMLButtonElement>();
+
+  useEffect(() => {
+    setShowCopyButton(copyable && ClipboardJS.isSupported());
+  }, [copyable]);
+
+  let copyButton: JSX.Element | undefined;
+
+  if (showCopyButton) {
+    let buttonVariant: ButtonVariant;
+
+    switch (variant) {
+      case Variant.Default:
+        buttonVariant = ButtonVariant.Default;
+        break;
+      case Variant.Dark:
+        buttonVariant = ButtonVariant.Dark;
+        break;
+    }
+
+    const trigger = (
+      <Button
+        ref={setButtonRef}
+        variant={buttonVariant}
+        className={buttonStyle}
+        onClick={() => setCopied(true)}
+      >
+        <CopyIcon size="large" className={iconStyle} />
+        Copy
+      </Button>
+    );
+
+    let tooltipVariant: TooltipVariant;
+
+    switch (variant) {
+      case Variant.Default:
+        tooltipVariant = TooltipVariant.Dark;
+        break;
+      case Variant.Dark:
+        tooltipVariant = TooltipVariant.Light;
+        break;
+    }
+
+    copyButton = (
+      // `Tooltip` forces its trigger to be `position: relative` :/
+      <div className={buttonWrapperStyle}>
+        <Tooltip
+          open={copied}
+          variant={tooltipVariant}
+          align={Align.Top}
+          justify={Justify.Middle}
+          trigger={trigger}
+          triggerEvent={TriggerEvent.Click}
+        >
+          Copied!
+        </Tooltip>
+      </div>
+    );
+  }
+
+  useEffect(() => {
+    if (!buttonRef) {
+      return;
+    }
+
+    const clipboard = new ClipboardJS(buttonRef, {
```

Comment:
Every instance of ClipboardJS [creates new listeners](https://github.com/zenorocha/clipboard.js/blob/master/src/clipboard.js#L18). We should destroy and cleanup a previous instance before we create a new one.